transaction integration test fails when run after other tests

**Describe the bug**
The transaction test will fail if it is run after the single-path and subscribe tests.

**To Reproduce**
Steps to reproduce the behavior:
1. make kind
2. onit run test single-path
3. onit run test subscribe
4. onit run test transaction

**Expected behavior**
Tests should all report PASS:

**Logs, text or screenshots**
Rays-MacBook:onos-config ray$ onit run test transaction
 ✓ Starting test job: test-2226283700 
=== RUN   transaction
--- FAIL: transaction (0.15s)
    transactiontest.go:91: 
        	Error Trace:	transactiontest.go:91
        	Error:      	Received unexpected error:
        	            	rpc error: code = Unknown desc = Can't remove last entry for device-1036654090-1.0.0%!(EXTRA *errors.errorString=Duplicate change ID 'l^M�����R��[�k�4\' in config)
        	Test:       	transaction
        	Messages:   	Rollback returned an error
    transactiontest.go:92: 
        	Error Trace:	transactiontest.go:92
        	Error:      	Expected value not to be nil.
        	Test:       	transaction
        	Messages:   	Response for rollback is nil
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x11cdbbd]

goroutine 33 [running]:
testing.tRunner.func1(0xc0002be900)
	/usr/local/go/src/testing/testing.go:830 +0x392
panic(0x1316260, 0x21b6660)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/onosproject/onos-config/test/integration.TestTransaction(0xc0002be900)
	/go/src/github.com/onosproject/onos-config/test/integration/transactiontest.go:93 +0x8ad
testing.tRunner(0xc0002be900, 0x1523390)
	/usr/local/go/src/testing/testing.go:865 +0xc0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:916 +0x35a

--- a/pkg/manager/rollbackconfig.go
+++ b/pkg/manager/rollbackconfig.go
@@ -46,7 +46,8 @@ func (m *Manager) RollbackTargetConfig(configname string) (change.ID, error) {
 func computeRollback(m *Manager, target string, configname string) (change.ID, map[string]*change.TypedValue, []string, error) {
 	id, err := m.ConfigStore.RemoveLastChangeEntry(store.ConfigName(configname))
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("Can't remove last entry for %s, %s", configname, err.Error())
+		return nil, nil, nil, fmt.Errorf("Can't remove last entry on target %s in config %s, %s",
+			target, configname, err.Error())
 	}
 	previousValues := make([]*change.ConfigValue, 0)
 	deletes := make([]string, 0)

--- a/pkg/manager/rollbackconfig.go
+++ b/pkg/manager/rollbackconfig.go
@@ -31,6 +31,7 @@ func (m *Manager) RollbackTargetConfig(configname string) (change.ID, error) {
 	log.Infof("Rolling back last change on config %s for target %s", configname, targetID)
 	id, updates, deletes, err := computeRollback(m, targetID, configname)
 	if err != nil {
+		log.Errorf("Error on rollback: %s", err.Error())
 		return nil, err
 	}
 	changeID, err := m.computeAndStoreChange(updates, deletes)
@@ -45,7 +46,7 @@ func (m *Manager) RollbackTargetConfig(configname string) (change.ID, error) {
 func computeRollback(m *Manager, target string, configname string) (change.ID, map[string]*change.TypedValue, []string, error) {
 	id, err := m.ConfigStore.RemoveLastChangeEntry(store.ConfigName(configname))
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf(fmt.Sprintf("Can't remove last entry for %s", configname), err)
+		return nil, nil, nil, fmt.Errorf("Can't remove last entry for %s, %s", configname, err.Error())
 	}
 	previousValues := make([]*change.ConfigValue, 0)
 	deletes := make([]string, 0)

--- a/pkg/northbound/gnmi/set.go
+++ b/pkg/northbound/gnmi/set.go
@@ -244,7 +244,7 @@ func listenForDeviceResponse(changes mapNetworkChanges, target string, name stor
 	mgr := manager.GetManager()
 	respChan, ok := mgr.Dispatcher.GetResponseListener(topocache.ID(target))
 	if !ok {
-		log.Infof("Device %s not properly registered, not waiting for southbound confirmation", target)
+		log.Infof("Device %s not properly registered, not waiting for southbound confirmation ", target)
 		return nil
 	}
 	//blocking until we receive something from the channel or for 5 seconds, whatever comes first.
@@ -252,7 +252,7 @@ func listenForDeviceResponse(changes mapNetworkChanges, target string, name stor
 	case response := <-respChan:
 		switch eventType := response.EventType(); eventType {
 		case events.EventTypeAchievedSetConfig:
-			log.Info("Set is properly configured", response.ChangeID())
+			log.Info("Set is properly configured ", response.ChangeID())
 			return nil
 		case events.EventTypeErrorSetConfig:
 			//Removing previously applied config

--- a/pkg/store/configuration.go
+++ b/pkg/store/configuration.go
@@ -128,14 +128,13 @@ func CreateConfiguration(deviceName string, version string, deviceType string,
 	configName := deviceName + "-" + version
 
 	//Look for duplicates in the changeId - do not sort
-	changesStrings := make([]string, len(changes))
-	for idx, c := range changes {
-		prevCh := strings.Join(changesStrings, ",")
-		if strings.Contains(prevCh, B64(c)) {
-			return nil, fmt.Errorf("Duplicate change ID '%s' in config",
-				c)
+	var previousChange string
+	for _, c := range changes {
+		if previousChange == B64(c) {
+			return nil, fmt.Errorf("Duplicate last change ID '%s' in config", B64(c))
+
 		}
-		changesStrings[idx] = B64(c)
+		previousChange = B64(c)
 	}
 
 	deviceConfig := Configuration{

